src/test/java/org/jitsi/meet/test/base/AbstractParticipantHelper.java

```diff
@@ -0,0 +1,177 @@
+     * if it hasn't been started.
+     */
+    public void waitForParticipant1ToJoinMUC()
+    {
+        if (participant1 == null)
+        {
+            participant1
+                = ParticipantFactory.getInstance()
+                    .createParticipant("web.participant1");
+        }
+
+        if (participant1.isHungUp())
+        {
+            // join room
+            participant1.joinConference(currentRoomName);
+        }
+
+        MeetUtils.waitForParticipantToJoinMUC(participant1.getDriver(), 15);
+    }
+
+    /**
+     * Setup helper by generating the room name to use.
+     */
+    public void setup()
+    {
+        this.currentRoomName
+            = "torture" + String.valueOf((int)(Math.random()*1000000));
+    }
+
+    /**
+     * Cleans up by quiting all participants.
+     */
+    public void cleanup()
+    {
+        quitParticipant(participant1);
+        quitParticipant(participant2);
+        quitParticipant(participant3);
+    }
+
+    /**
+     * Quits a participant if initialized.
+     * @param participant the participant to quit.
+     */
+    private void quitParticipant(Participant<WebDriver> participant)
+    {
+        if (participant == null)
+            return;
+
+        participant.quit();
+    }
+
+    /**
+     * Waits until {@code secondParticipant} has joined the conference (its ICE
+     * connection has completed and has it has sent and received data).
+     */
+    public void waitForSecondParticipantToConnect()
+    {
+        if (participant2 == null)
+        {
+            participant2 = ParticipantFactory.getInstance()
+                .createParticipant("web.participant2");
+        }
+
+        if (participant2.isHungUp())
+        {
+            participant2.joinConference(currentRoomName);
+        }
+
+        MeetUtils.waitForParticipantToJoinMUC(participant2.getDriver(), 10);
+        MeetUtils.waitForIceConnected(participant2.getDriver());
+        MeetUtils.waitForSendReceiveData(participant2.getDriver());
+
+        TestUtils.waitMillis(5000);
+    }
+
+    /**
+     * Starts the owner, if it isn't started and hangups all other participants.
+     */
+    public void closeAllParticipantsExceptTheOwner()
+    {
+        waitForParticipant1ToJoinMUC();
+
+        if (participant2 != null && !participant2.isHungUp())
+        {
+            participant2.hangUp();
+        }
+
+        if (participant3 != null && !participant3.isHungUp())
+        {
+            participant3.hangUp();
+        }
+    }
+
+    /**
+     * Waits until {@code thirdParticipant} has joined the conference (its ICE
+     * connection has completed and has it has sent and received data).
+     */
+    public void waitForThirdParticipantToConnect()
+    {
+        if (participant3 == null)
+        {
+            participant3 = ParticipantFactory.getInstance()
+                .createParticipant("web.participant3");
+        }
+
+        if (participant3.isHungUp())
```

Comment:
It reads weird or is hard to understand. If participant "is hung up" then join, but otherwise there's no need to call joinConference like it would be doing that automatically ? What if it was not called previously ? For me it would read easier if "joinConference" would be doing the "isHungUp" check.